node/src/components/consensus/protocols/highway.rs

```diff
@@ -734,69 +796,90 @@ where
+                        vec![ProtocolOutcome::CreatedTargetedMessage(
+                            HighwayMessage::NewVertex(vv.into()).serialize(),
+                            sender,
+                        )]
+                    }
+                }
+            }
+            Ok(HighwayMessage::LatestStateRequest(their_index_panorama)) => {
                 trace!("received a request for the latest state");
                 let state = self.highway.state();
 
                 let create_message = |((vid, our_obs), their_obs): (
-                    (ValidatorIndex, &Observation<C>),
-                    &Observation<C>,
+                    (ValidatorIndex, &IndexObservation),
+                    &IndexObservation,
                 )| {
                     match (our_obs, their_obs) {
-                        (our_obs, their_obs) if our_obs == their_obs => None,
+                        (our_obs, their_obs) if our_obs == their_obs => vec![],
 
-                        (Observation::None, Observation::None) => None,
+                        (IndexObservation::None, IndexObservation::None) => vec![],
 
-                        (Observation::Faulty, _) => state.maybe_evidence(vid).map(|evidence| {
-                            HighwayMessage::NewVertex(Vertex::Evidence(evidence.clone()))
-                        }),
+                        (IndexObservation::Faulty, _) => state
+                            .maybe_evidence(vid)
+                            .map(|evidence| {
+                                HighwayMessage::NewVertex(Vertex::Evidence(evidence.clone()))
+                            })
+                            .into_iter()
+                            .collect(),
 
-                        (_, Observation::Faulty) => {
+                        (_, IndexObservation::Faulty) => {
+                            let dependency = Dependency::Evidence(vid);
                             let uuid = thread_rng().next_u64();
-                            trace!(?uuid, "requesting evidence");
-                            Some(HighwayMessage::RequestDependency(
-                                uuid,
-                                Dependency::Evidence(vid),
-                            ))
+                            debug!(?uuid, "requesting evidence");
+                            vec![HighwayMessage::RequestDependency(uuid, dependency)]
                         }
 
-                        (Observation::None, Observation::Correct(hash)) => {
-                            let uuid = thread_rng().next_u64();
-                            let dependency = Dependency::Unit(*hash);
-                            trace!(?uuid, ?dependency, "requesting dependency");
-                            Some(HighwayMessage::RequestDependency(uuid, dependency))
+                        (IndexObservation::None, IndexObservation::Correct(their_seq_number)) => {
+                            self.batch_request(vid, 0, *their_seq_number)
```

Comment:
We don't actually have sequence number 0.
Maybe `batch_request` should take the lowest number we _don't_ have instead of the highest one we have.